nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -13,9 +14,112 @@
+        return np.arange(self.get_movie_frame_count()) / self.get_movie_fps()
+
+    def get_movie_fps(self):
+        """Return the internal frames per second (fps) for a movie file"""
+        assert self.vc.isOpened(), self._movie_open_msg
+        if int(cv2.__version__.split(".")[0]) < 3:
+            return self.vc.get(cv2.cv.CV_CAP_PROP_FPS)
+        return self.vc.get(cv2.CAP_PROP_FPS)
+
+    def get_frame_shape(self) -> Tuple:
+        """Return the shape of frames from a movie file."""
+        return self.frame.shape
+
+    def get_movie_frame_count(self):
+        """Return the total number of frames for a movie file."""
+        assert self.vc.isOpened(), self._movie_open_msg
+        if self.stub:
+            # if stub the assume a max frame count of 10
+            return 10
+        if int(cv2.__version__.split(".")[0]) < 3:
+            count = self.vc.get(cv2.cv.CV_CAP_PROP_FRAME_COUNT)
+        else:
+            count = self.vc.get(cv2.CAP_PROP_FRAME_COUNT)
+        return int(count)
+
+    @property
+    def current_frame(self):
+        return self._current_frame
+
+    @current_frame.setter
+    def current_frame(self, frame_no):
+        assert self.vc.isOpened(), self._movie_open_msg
+        if int(cv2.__version__.split(".")[0]) < 3:
+            set_arg = cv2.cv.CV_CAP_PROP_POS_FRAMES
+        else:
+            set_arg = cv2.CAP_PROP_POS_FRAMES
+        set_value = self.vc.set(set_arg, frame_no)
+        if set_value:
+            self._current_frame = frame_no
+        else:
+            raise ValueError(f"could not set frame no {frame_no}")
+
+    def get_movie_frame(self, frame_no: int):
+        """Return the specific frame from a movie."""
+        assert self.vc.isOpened(), self._movie_open_msg
+        assert frame_no < self.get_movie_frame_count(), "frame number is greater than length of movie"
+        self.current_frame = frame_no
+        success, frame = self.vc.read()
+        self.current_frame = 0
+        if success:
+            return frame
+        elif frame_no > 0:
+            return np.nan * np.ones(self.get_frame_shape())
+
+    def get_movie_frame_dtype(self):
+        """Return the dtype for frame in a movie file."""
+        return self.frame.dtype
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        assert self.vc.isOpened(), self._movie_open_msg
+        if self._current_frame < self.frame_count:
+            self.current_frame = self._current_frame
+            success, frame = self.vc.read()
+            self._current_frame += 1
+            self.vc.release()
+            if success:
+                return frame
+            else:
+                raise StopIteration
+        else:
+            self.vc.release()
+            raise StopIteration
+
+    def __enter__(self):
+        self.vc = cv2.VideoCapture(self.file_path)
+        return self
+
+    def __exit__(self, *args):
+        self.vc.release()
+
+    def __del__(self):
+        self.vc.release()
```

Comment:
```suggestion
    def release(self):
        self.vc.release()

    def __exit__(self, *args):
        self.vc.release()

    def __del__(self):
        self.vc.release()
```

I'd suggest adding an additional command to release the internal attribute at user discretion. Both the other methods act more or less automatically, typically hidden to the user.

Reply:
yeah this is more user friendly. I've made the changes